### PR TITLE
Local Source - qol, cleanup and cover related fixes

### LINKED ISSUE #3394: [Bug] Problem with local manga and sd card
### Device information
* Tachiyomi version: 0.9.2
* Android version: 7.1.2
* Device: Kindle Fire HD 10 (2017)

## Steps to reproduce
1. copy manga folders to local folder
2. refresh/close app/restart device

### Expected behavior
local manga should show up

### Actual behavior
local manga doesn't show up

### Other details

I've looked over a bunch of other topics before making one and haven't seen a solution, i see [this link](https://tachiyomi.org/help/guides/reading-local-manga/) posted but it doesn't apply to me as all my folders are already correct and were used in tachiyomi app previously, they were all on my sd card and are didn't get modified. They also work fine if copied directly to internal storage so its not an issue with the naming.

Everything was working fine with the app but amazon's skin finally annoyed me enough and i wanted a more stock android so i flashed the nougat rom, so i assume that this is some issue with nougat that is screwing things up.

But it's not as if the sd card isn't being read/written to, i can download from regular sources to the sd card and they appear under the library fine, it is only the local folder that doesn't seem to want to cooperate.

### LINKED ISSUE #5319: [Feature Request] Generate cover.jpg in the sd card local directory instad of always saving it at /internal/Tachiyomi/local
## Why/User Benefit/User Problem
Right now when a user adds a local manga at /sdcard/Tachiyomi/local Tachiyomi creates a new directory at /internal/Tachiyomi/local with the same name as the manga directory on the sdcard and saves the cover.jpg there. If a user deletes a local manga from the sdcard without deleting it from the internal storage the manga will still be shown in Tachiyomi just without the chapters. To really remove a manga a user has to search through their internal storage  and delete the manga from there as well.

## What/Requirements
save the cover.jpg into the directory were the manga is saved instead of saving it to the path /internal/Tachiyomi/local

### LINKED ISSUE #6737: Set as cover requires app restart to take effect
### Steps to reproduce

1. In a manga stored in a folder on the SD card long-press and select "Set as cover".
2. Refresh manga listing by dragging down to refresh.
3. In library view of titles refresh by either clicking refresh icon or dragging down to refresh.

### Expected behavior

As soon as an image is selected with set as cover, the library view should update for that title.

### Actual behavior

The original image continues to be displayed for both reproduction steps 2 and 3, the new image is not displayed until after selecting the recent apps and swiping away Tachiyomi to force-close it. After restarting the app, the cover previously selected and set will now be displayed.

### Crash logs

_No response_

### Tachiyomi version

0.13.1

### Android version

Android 9

### Device

Samsung A70

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #6751: [Local Manga] Can not set cover image
### Steps to reproduce

Add a manga to local that does not include a cover image.
In Tachiyomi browse the manga select a page, hold on screen wait for the options.
[Set as cover]
[Share]
[Save]

Select [Set as cover]
"Use this image as cover art?"
OK

the prompt 'Cover Set' breifly shows up 

### Expected behavior

The cover should be set in the Chapter Select or browse.

### Actual behavior

nothing is there, neither in the chapter select or in the browse.

### Crash logs

_No response_

### Tachiyomi version

Stable 0.13.1

### Android version

Android 12

### Device

Pixel 6

### Other details

This may be similar to issue #6737
but is not the same. Difference being that the cover IS NEVER set even after restart if the manga didn't have any cover.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have tried the [troubleshooting guide](https://tachiyomi.org/help/guides/troubleshooting/).
- [X] I have updated the app to version **[0.13.1](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I have updated all installed extensions.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #7021: creat nomedia with cover image
### Describe your suggested feature

With the cover.jpg image there should be a .nomedia file
other wise my gellary will get full.

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] If this is an issue with an extension, I should be opening an issue in the [extensions repository](https://github.com/tachiyomiorg/tachiyomi-extensions/issues/new/choose).
- [X] I have updated the app to version **[0.13.4](https://github.com/tachiyomiorg/tachiyomi/releases/latest)**.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by FourTOne5**
Closes #7021 
Closes #5319
Fixes #6751 
Fixes #3394
Fixes #6737

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt [hunk 1]
```diff
@@ -30,176 +31,233 @@ import logcat.LogPriority
...
+            inputStream.use { input ->
+                coverFile.outputStream().use { output ->
+                    input.copyTo(output)
                 }
             }
-            manga.thumbnail_url = cover.absolutePath
-            return cover
+
+            // Create a .nomedia file
+            File(mangaDir.absolutePath, NOMEDIA_NAME).createNewFile()
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt:80]: You can use `DiskUtil.createNoMediaFile(...)`

File: app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt [hunk 2]
```diff
@@ -30,176 +31,233 @@ import logcat.LogPriority
...
-                it.isFile && ImageUtil.isImage(it.name) { it.inputStream() }
-            }
+        private fun getCoverFile(mangaUrl: String, baseDirsFile: Sequence<File>): File? {
+            return baseDirsFile
+                // Filter out ones that are not related to the manga and is not a directory
+                .filter { it.isDirectory && it.name == mangaUrl }
+                // Get all the files inside the manga folders
+                .flatMap { it.listFiles().orEmpty().toList() }
+                // Get all file whose names start with 'cover'
+                .filter { it.isFile && it.nameWithoutExtension.lowercase() == "cover" }
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt:93]: Can uses `.equals("cover", ignoreCase = true)`

File: app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt [hunk 3]
```diff
@@ -211,13 +269,6 @@ class LocalSource(private val context: Context) : CatalogueSource, UnmeteredSour
                     }
                     date_upload = chapterFile.lastModified()
 
-                    val format = getFormat(chapterFile)
-                    if (format is Format.Epub) {
-                        EpubFile(format.file).use { epub ->
-                            epub.fillChapterMetadata(this)
-                        }
-                    }
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt:219]: Don't we still need this?

**Review by arkon — CHANGES_REQUESTED** [review]

File: app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt (hunk 3 above)
Comment (FourTOne5) [reply app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt:219]: The modified manga only stays in memory. Could move this in `getMangaDetails` if you want

Comment (FourTOne5) [reply app/src/main/java/eu/kanade/tachiyomi/source/LocalSource.kt:219]: Actually I just realized this is a different method

**Pull request merged**